dao/src/main/java/com/epam/esm/dao/TagDao.java

```diff
@@ -0,0 +1,14 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.domain.entity.Tag;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface TagDao extends DBOperationCRD<Tag> {
+
+    Optional<Tag> readByTagName(String name);

```

Comment:
You don't need to repeat entity name in every method name. It's obvious that **TAG**dao works with tags.

Reply:
thanks. renamed